sdk/identity/Azure.Identity/src/AzureApplicationCredential.cs

```diff
@@ -37,8 +37,16 @@ public AzureApplicationCredential(AzureApplicationCredentialOptions options) : t
 
         internal AzureApplicationCredential(AzureApplicationCredentialOptions options, EnvironmentCredential environmentCredential = null, ManagedIdentityCredential managedIdentityCredential = null)
         {
+            EnvironmentCredentialOptions environmentCredentialOptions = new EnvironmentCredentialOptions
```

Comment:
[Note] In this case (and also in EnvironmentCredential.cs), I'm cloning the values from the base `TokenCredentialOptions` when creating the new `EnvironmentCredentialOptions`. This seems to technically be necessary to not lose the authority host, PII logging, and transport settings, but I don't see anyone else doing it anywhere, so this might be overkill. (In all other places, I see creation of brand new options and discarding the base class properties.)

Reply:
Thanks @trevorlacey-msft  - Would you mind linking to one of the examples you are referring to where we seem to lose the base class property values?

Reply:
I was referring to the `TokenCredentialOptions` that are passed into `DefaultAzureCredentialFactory` [here](https://github.com/Azure/azure-sdk-for-net/blob/d1623b7372d184996feb38dde39b1586ff746aed/sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs#L11). In most of the `CreateXXXCredential()` functions in that file, we create brand new sub-options classes without the properties from the original `TokenCredentialOptions`.

Some of the "missing" options appear to only be used by the actual credentials, so they're not actually missing. Not all, though.
- **[BAD]** `TokenCredentialOptions.IsLoggingPIIEnabled` is used all over the place, so `DefaultAzureCredentialOptions` (incorrectly?) drops this property when creating other sub-options
- **[OK]** `TokenCredentialOptions.AuthorityHost` is currently only used by `CredentialPipeline`, and it is already being correctly plumbed
- **[OK]** `ClientOptions.Transport` is never read in prod code--only in perf tests
- **[OK]** `ClientOptions.Policies` is never read--maybe this functionality has been superceded by `RequestContext.Policies`?
- **[OK]** `ClientOptions.Retry` is never read
- **[OK]** `ClientOptions.Diagnostics` is used indirectly only by `CredentialPipeline` (via `ClientDiagnostics`), and it is already being correctly plumbed

Maybe this means when I create my new `EnvironmentCredentialOptions` that I only need to copy over `IsLoggingPIIEnabled`. What do you think?

Reply:
Thanks for the context - yes, I think we should be properly transferring these properties to the sub-options. However, rather than doing it explicitly in each credential, I think we should centralize the logic somewhere. For example, we could create a constructor on `TokenCredentialOptions` that takes in a `TokenCredentialOptions` where this is done.

If you don't want to open that can of worms as part of this PR, I'll create a separate tracking issue to address it in a separate PR.

Reply:
I'd rather leave a change like that to the experts as a separate PR. :-)